### ISSUE TITLE
Service module doesn't fail when "status" command does not exists

### ISSUE THREAD

**Issue by willybarro** [ansible/ansible-modules-core#849]
##### Issue Type:

Bug Report
##### Ansible Version:

1.8.4
##### Environment:

Debian GNU/Linux 7
##### Summary:

While using the service module, if the called init script (/etc/init.d/<service-name>) doesn't have the "status" command, ansible will return "changed", even though nothing happened and it should have failed. This is particularly bad for **"restarted"** and **"reloaded"** states, because if ansible can't check the status of a service, it'll simply try to start the service again without trying to stop it before, thus, giving the false sensation that something happened.

Verbose mode ("-vvvv") also doesn't help to debug that, because the stderr produced by the script is silently ignored on the service module.

**My usecase:**
We had a php5-fpm init.d script that didn't had the status command (now it's fixed, but that's not the case). 
Every deploy needs a reload on php5-fpm to refresh caches. Ansible said it had sucessfully reloaded the service, but it didn't and opcaches was not refreshed, then our changes never went live.
##### Steps To Reproduce:

Run the playbook below with the example init.d script installed.
###### Playbook

```

---
- name: "Reproducing bug"
  hosts: "33.33.33.100"
  serial: "1"
  tasks:
    - name: Reload bug ansible test initd service
      sudo: yes
      service: name="bug-ansible-test-initd" state="reloaded"
```
###### Example init.d script

Copy the script below to /etc/init.d/ and register it as a service (use rcconf for this).

https://gist.github.com/willybarro/cf9ce794caab51a1b335
##### Expected Results:

```
failed: [33.33.33.10] => {"failed": true}
msg: init.d script does not have the "status" command
```
##### Actual Results:

```
changed: [33.33.33.100] => {"changed": true, "name": "bug-ansible-test-initd", "state": "started"}

PLAY RECAP ********************************************************************
33.33.33.100               : ok=2    changed=1    unreachable=0    failed=0
```

**Comment by bcoca** [comment]
So sadly 'status' is not part of the standard init scripts are supposed to follow, though most good init script writers (yes, they exist) do add it. The service module adds the 'pattern' argument for those scripts that do not implement status (as documented in the module) or for those that don't implement it correctly.

**Comment by willybarro** [comment]
Yes, and I found that solution. But don't you think that the module should check for that default command and at least provide a warning or a more clear verbose output in those cases? It was really hard to debug that. Oh... And I'll be happy to provide a PR for that if you guys think it makes sense.

**Comment by willybarro** [comment]
I was thinking in something like checking if the stderr contains "Usage:", or, if rc = 1 and stderr contains anything, it should output the error on the verbose output or throw an error.

**Comment by bcoca** [comment]
it makes sense, but i need to look at code as iirc the init script standard does give the return codes specific meanings.

**Comment by ansibot** [comment]
@grossws, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@grossws, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
This issue was migrated to https://github.com/ansible/ansible/issues/30172
